crates/holo_hash/src/hash_type/primitive.rs

```diff
@@ -178,6 +180,18 @@ impl HashTypeSync for Header {}
 // A DnaHash is a hash of the DnaDef, which excludes the wasm bytecode
 impl HashTypeSync for Dna {}
 
+// We don't know what external data might be getting hashed but typically it
+// would be small, like a reference to something in an external system such as
+// a hash in a different DHT, an IPFS hash or a UUID, etc.
+/// External hashes have a DHT location and hash prefix like all other native
+/// holochain hashes but are NOT found/fetchable on the DHT.
+/// External hashing makes no assumptions about the data that was digested to
+/// create the hash so arbitrary bytes can be passed in.
+/// It is valid to EITHER use an existing 32 byte hash/data as literal bytes
+/// for an external hash (literal+prefix, no data loss) OR digest arbitrary
+/// data into an external hash (support all data, opaque result).
+impl HashTypeSync for External {}
```

Comment:
Does this make sense. If the data could be arbitrarily large shouldn't this be `HashTypeAsync`?

Reply:
After reading the other PR I think this should by `Async` because the host function is using blake directly anyway. That way no one will accidentally hash this from within an async context but it will still be possible from the host function which is always called from within a `spawn_blocking` anyway.

Reply:
alternately, we could put a cap on the max size of this data, say 8K bytes or something that won't get us into async danger, and if the source data is larger than that, they'll have to generic hash it first?